code/modules/games/cards/lotto.dm

```diff
@@ -0,0 +1,131 @@
+/obj/item/toy/lotto_ticket
+	name = "scratch-off lotto ticket"
+	desc = "A scratch-off lotto ticket."
+	icon = 'icons/obj/toy.dmi'
+	w_class = W_CLASS_TINY
+	var/obj/item/toy/lotto_ticket/revealed = 0
+	var/obj/item/toy/lotto_ticket/iswinner = 0
+	var/obj/item/toy/lotto_ticket/ticket_price
+	var/obj/item/toy/lotto_ticket/winnings = 0
+
+/obj/item/toy/lotto_ticket/New()
+	..()
+	pixel_y = rand(-8, 8) * PIXEL_MULTIPLIER
+	pixel_x = rand(-9, 9) * PIXEL_MULTIPLIER
+
+/obj/item/toy/lotto_ticket/proc/scratch(var/input_price)
+	var/list/prizelist = list(100000,50000,10000,5000,1000,500,250,100,50,20,10,5,4,3,2,1)
+	var/list/problist = list(0.0001, 0.0002, 0.001, 0.002, 0.01, 0.02, 0.04, 0.2, 1, 2.5, 5, 10, 12.5, 17, 20, 25)
+	var/tuning_value = 1/5 //Used to adjust expected values.
+	var/profit = 0
+	if(istype(src,/obj/item/toy/lotto_ticket/supermatter_surprise))
+		input_price = 5	//Sets the prize table equal to the tier-1 card but with a guaranteed win
+	while(profit == 0)
+		for(var/prize = 1 to problist.len)
+			if(prob(problist[prize]))
+				return(prizelist[prize]*input_price*tuning_value)
```

Comment:
style

`return prizelist[prize]*input_price*tuning_value`

Reply:
Resolved in latest commit